src/Initializer.js

```diff
@@ -1,289 +1,407 @@
 'use strict'
 
+var beautify = require('js-beautify').js_beautify
 /**
  * Initializer
  */
 class Initializer {
 
   /**
-   * Compile (static)
+   * constructor
    */
-  static compile (schema) {
-    let initializer = new Initializer(schema)
-    initializer.parse()
-    return initializer.compile()
+  constructor (schema, options) {
+    Object.assign(this, options || {})
```

Comment:
Actually, n/m for this particular constructor -- @christiansmith pointed out, this is a singleton, and not really concerned with its in-memory representation.

I highly recommend reading the article, though, it's very surprising and interesting.